src/cpp/session/modules/SessionFind.hpp

```diff
@@ -28,6 +31,70 @@ core::json::Object findInFilesStateAsJson();
 
 core::Error initialize();
 
+// helper class used to process file replacements
+class Replacer : public boost::noncopyable
+{
+   core::Error completeReplace(const boost::regex* searchRegex, const std::string* replaceRegex,
+                               size_t matchOn, size_t matchOff, std::string* line,
+                               size_t* pReplaceMatchOff);
+
+   core::Error replaceRegexIgnoreCase(size_t matchOn, size_t matchOff, std::string* line,
+                                      const std::string* findRegex, const std::string* replaceRegex,
+                                      size_t* pReplaceMatchOff);
+
+   core::Error replaceRegexWithCase(size_t matchOn, size_t matchOff, std::string* line,
+                                    const std::string* findRegex, const std::string* replaceRegex,
+                                    size_t* pReplaceMatchOff);
+
+public:
+   explicit Replacer(bool ignoreCase) :
+      ignoreCase_(ignoreCase)
+   {
+   }
+
+   void replaceLiteralWithLiteral(size_t matchOn, size_t matchOff, std::string* line,
+                                  const std::string* replaceLiteral,
+                                  size_t* pReplaceMatchOff)
+   {
+      *line = line->replace(matchOn, (matchOff - matchOn), *replaceLiteral);
+      *pReplaceMatchOff = matchOn + replaceLiteral->size();
+   }
+
+   core::Error replaceLiteralWithRegex(size_t matchOn, size_t matchOff, std::string* line,
+                                const std::string* replaceRegex,
+                                size_t* pReplaceMatchOff)
+   {
+      std::string find(line->substr(matchOn, matchOff));
+      return (replaceRegexWithRegex(matchOn, matchOff, line, &find,
+                                    replaceRegex, pReplaceMatchOff));
+   }
+
+   core::Error replaceRegexWithLiteral(size_t matchOn, size_t matchOff, std::string* line,
```

Comment:
On second though, I made the requested changes and remove the replaceRegex flag. Since this involved several changes, opening project card for Maria to further review.